### ISSUE TITLE
Crafting model update please sort it into files and translate it into english

### ISSUE THREAD

**Issue by Nomoos** [Nomoos/BlueMarble.Design#99]
📐 Formální model výroby (bez pevných thresholdů)
Značení

s … aktuální skill hráče (v dané disciplíně)

r … doporučený/požadovaný skill receptu (jeho náročnost)

x = s / r … relativní skill (bezrozměrné)

U(0,1) … rovnoměrné náhodné číslo na intervalu ⟨0,1⟩

clamp(z, a, b) … oříznutí do intervalu ⟨a, b⟩

1) Podmínka pro pokus o výrobu

„Beze skillu nemůžeš recept vyrábět.“

Pokus povolen jen pokud s > 0.
(Tj. hráč danou dovednost aspoň má – i kdyby nízkou.)

2) Pravděpodobnost selhání výroby p_fail(x)

„Od čtvrtinového skillu je šance na selhání minimální (téměř nulová), a při selhání nepřijdeš o materiál. Při velmi nízkém skillu selhání časté.“

Použijeme hladkou funkci, která:

pro x = 0 dává selhání ~100 %,

klesá k ~0 % u x ≥ 0.25,

zůstává velmi nízká i pro vyšší x.

𝑝
fail
(
𝑥
)
  
=
  
{
1
−
(
𝑥
0.25
)
𝛾
,
	
0
≤
𝑥
<
0.25

𝜀
⋅
𝑒
−
𝑘
 
(
𝑥
−
0.25
)
,
	
𝑥
≥
0.25
p
fail
	​

(x)=
⎩
⎨
⎧
	​

1−(
0.25
x
	​

)
γ
,
ε⋅e
−k(x−0.25)
,
	​

0≤x<0.25
x≥0.25
	​

Doporučené parametry (intuitivní chování):

𝛾
=
1.5
γ=1.5 (tvar křivky v nízkých skillech)

𝜀
=
0.05
ε=0.05 (zbytková „téměř nulová“ šance)

𝑘
=
2.0
k=2.0

Materiál při selhání:

𝑝
loss
(
𝑥
)
=
{
1
,
	
𝑥
<
0.25

0
,
	
𝑥
≥
0.25
p
loss
	​

(x)={
1,
0,
	​

x<0.25
x≥0.25
	​

(Tj. pod čtvrtinovým skillem selhání pálí materiál, od 0.25 výš už ne.)

3) Náhodné rozdělení kvality quality ∈ ⟨1,100⟩

„S rostoucím skillem roste šance na lepší kvalitu; s malým skillem je prakticky nulová šance dostat kvalitu > 30 %.“

Zavedeme S-křivku pro střed kvality a snižující se rozptyl se skill/faktorem x.
Nechť:

𝜇
(
𝑥
)
=
100
⋅
𝜎
 ⁣
(
𝑎
⋅
(
𝑥
−
1
)
)
,
𝜎
(
𝑧
)
=
1
1
+
𝑒
−
𝑧
μ(x)=100⋅σ(a⋅(x−1)),σ(z)=
1+e
−z
1
	​

𝜎
𝑄
(
𝑥
)
=
max
⁡
 ⁣
(
𝜎
min
⁡
,
 
𝜎
0
⋅
𝑒
−
𝑏
 
(
𝑥
−
0.25
)
)
σ
Q
	​

(x)=max(σ
min
	​

,σ
0
	​

⋅e
−b(x−0.25)
)

Doporučené parametry:

𝑎
=
3.0
a=3.0 (strmost S-křivky; 
𝑥
=
1
x=1 → 
𝜇
=
50
μ=50)

𝜎
0
=
25
σ
0
	​

=25, 
𝑏
=
1.2
b=1.2, 
𝜎
min
⁡
=
5
σ
min
	​

=5

Generování kvality (truncated normal):

Vygeneruj 
𝑍
∼
𝑁
(
𝜇
(
𝑥
)
,
 
𝜎
𝑄
(
𝑥
)
2
)
Z∼N(μ(x),σ
Q
	​

(x)
2
)

quality
=
clamp
(
round
(
𝑍
)
,
 
1
,
 
100
)
quality=clamp(round(Z),1,100)

Pozn.: Pro velmi malé x vychází 
𝜇
(
𝑥
)
μ(x) nízko (např. při x=0.5 je 
𝜇
≈
18
μ≈18), takže 
𝑃
(
quality
>
30
)
P(quality>30) je skutečně velmi malá.

4) Celý postup výroby (pseudo-algoritmus)
vstup: s, r  // skilly a náročnost receptu
x = s / r
if s <= 0:
    return "Nelze vyrábět (nemáš dovednost)."

u = U(0,1)
p_fail = (x < 0.25) ? (1 - (x/0.25)^γ) : (ε * exp(-k*(x - 0.25)))

if u < p_fail:
    // SELHÁNÍ
    loss = (x < 0.25) ? "materiál ztracen" : "materiál zachován"
    return { result: "fail", material: loss }
else:
    // ÚSPĚCH → GENERUJ KVALITU
    μ = 100 * (1 / (1 + exp(-a*(x - 1))))
    σQ = max(σ_min, σ0 * exp(-b*(x - 0.25)))
    Z = Normal(μ, σQ^2)
    quality = clamp(round(Z), 1, 100)
    return { result: "success", quality: quality }

5) Intuice & ladění

„Téměř nulová“ šance na selhání od 0.25: nastavuje 
𝜀
ε a 
𝑘
k.

Např. při x = 0.5: 
𝑝
fail
≈
0.05
⋅
𝑒
−
2
⋅
0.25
≈
0.029
p
fail
	​

≈0.05⋅e
−2⋅0.25
≈0.029 (≈3 %)

„Téměř nulová“ šance na >30 při malém skille: řídí 
𝑎
a a 
𝜎
𝑄
σ
Q
	​

.

Při x = 0.5: 
𝜇
≈
18
μ≈18, 
𝜎
𝑄
≈
18
σ
Q
	​

≈18 ⇒ pravděpodobnost nad 30 je malá.

Rychlost konvergence k vysoké kvalitě (80–100): zvětši 
𝑎
a (strmější S-křivka) nebo zrychli pokles 
𝜎
𝑄
σ
Q
	​

 (větší 
𝑏
b).

Hladkost: vše je spojité, žádné tvrdé schody.

6) Volitelné modifikátory (materiál, nástroje, prostředí)

Můžeš přidat additivní/multiplikativní bonusy do x (efektivní skill) bez rozbití modelu:

𝑥
′
=
𝑥
⋅
(
1
+
tool_mult
)
+
material_bonus
x
′
=x⋅(1+tool_mult)+material_bonus

Pak stejné vzorce použij s x' místo x.

Lepší nástroj: tool_mult = +0.10 (efektivně +10 % k x)

Špičkový materiál: material_bonus = +0.05 (přidá 0.05 k x)

Špatné podmínky: záporné bonusy

Krátký numerický příklad

r = 60 (recept středně těžký)

s = 30 ⇒ x = 0.5

p_fail ≈ 0.029 (≈3 %)

μ ≈ 18, σ_Q ≈ 18 ⇒ typicky vyjde kvalita kolem 10–30, nad 30 jen zřídka.

s = 60 ⇒ x = 1.0 → μ ≈ 50, σ_Q ≈ 12

s = 120 ⇒ x = 2.0 → μ ≈ 95, σ_Q ≈ 5 (vysoká a stabilní kvalita)
